### PR TITLE
Convert manpage to ReStructuredText format

### PR DESCRIPTION
**Pull request by jonner**
This should make the manpage a bit easier to maintain in a more human-readable way. libvirt uses the rst2man utility to manage its manpages and it seems to work quite well. It adds a build-time dependency on python-docutils, but that doesn't seem too onerous.

### CODE REVIEW

**Comment by jonner** [comment]
FYI: here's the (whitespace-agnostic) difference between the rendered output of the old manpage to the new manpage using the following command:

`diff -uw manpage.old manpage.new`

Most of the differences are due to slight indentation changes causing word-wrapping in different places.

```diff
--- manpage.old	2021-10-01 14:58:40.285371552 -0500
+++ manpage.new	2021-10-01 14:58:32.381326077 -0500
@@ -1,16 +1,20 @@
-mdevctl(8)                                     System Manager's Manual                                    mdevctl(8)
+MDEVCTL(8)                                                                                                MDEVCTL(8)
 
 NAME
-       mdevctl, lsmdev - Mediated device management utility
+       mdevctl - Mediated device management utility
 
 SYNOPSIS
        mdevctl {COMMAND} [OPTIONS...]
 
+       lsmdev [OPTIONS...]
+
 DESCRIPTION
        mdevctl is a utility for managing and persisting devices in the mediated device device framework of the Linux
        kernel.  Mediated devices are sub-devices of a parent device (ex. a vGPU) which can  be  dynamically  created
        and potentially used by drivers like vfio-mdev for assignment to virtual machines.
 
+       lsmdev is an alias for mdevctl list.
+
 OPTIONS
        The following options are understood:
 
@@ -62,48 +66,47 @@
 
        define DEVICESPEC
            Define a config for an mdev device, identified either by an UUID (if the device already  exists),  or  by
-           the parent device and either the type or a JSON configuration file, and, optionally, the UUID. If no UUID
-           is specified, one is autogenerated and printed. If no file is used, -a|--auto may be used to specify that
-           the device should be started automatically.
-
-       list
-           List  mdev devices. With no options, currently running devices are listed.  With -d|--defined, previously
-           defined devices are listed.  Can be restricted to list only devices for a  given  parent  or  UUID.  With
-           --dumpjson  output  is  provided in machine readable JSON format.  When a UUID is provided and the output
-           results in a single device, the JSON output format is compatible with the configuration file format.
+              the  parent  device and either the type or a JSON configuration file, and, optionally, the UUID. If no
+              UUID is specified, one is autogenerated and printed. If no file is used,  -a|--auto  may  be  used  to
+              specify that the device should be started automatically.
+
+       list   List  mdev  devices. With no options, currently running devices are listed.  With -d|--defined, previ‐
+              ously defined devices are listed.  Can be restricted to list only devices for a given parent or  UUID.
+              With  --dumpjson  output is provided in machine readable JSON format.  When a UUID is provided and the
+              output results in a single device, the JSON output format is compatible with  the  configuration  file
+              format.
 
        modify DEVICESPEC
-           Modify the configuration for an mdev device, identified via its UUID and optionally its parent.  Type and
-           startup  mode  (auto or manual) can be modified by this command.  Attributes can be added or deleted. At‐
-           tributes to be deleted must be specified by their index; if an attribute is specified without  an  index,
-           it is appended at the end of the attribute list.  Running devices are unaffected by this command; changes
-           in the configuration are applied the next time the device is started.
+              Modify  the configuration for an mdev device, identified via its UUID and optionally its parent.  Type
+              and startup mode (auto or manual) can be modified  by  this  command.   Attributes  can  be  added  or
+              deleted. Attributes to be deleted must be specified by their index; if an attribute is specified with‐
+              out an index, it is appended at the end of the attribute list.  Running devices are unaffected by this
+              command; changes in the configuration are applied the next time the device is started.
 
        start DEVICESPEC
-           Start a mediated device. This command can be used to start either a previously-defined device or a newly-
-           created transient device.
+              Start  a  mediated  device.  This command can be used to start either a previously-defined device or a
+              newly-created transient device.
 
-           If  the UUID and optional parent argument matches an existing device definition, then the existing device
-           will be started. It is an error to specify a device type that conflicts with the existing device  defini‐
-           tion.
+              If the UUID and optional parent argument matches an existing device definition, then the existing  de‐
+              vice  will be started. It is an error to specify a device type that conflicts with the existing device
+              definition.
 
            If  the  UUID  argument  is omitted or if the specified UUID and parent does not match an existing device
-           definition, a new transient device will be started.  If the UUID is omitted, a new UUID will be automati‐
-           cally  generated.  When starting a new transient device, the parent and device type must be specified.  A
-           --jsonfile may replace the --type specification and also include additional attributes in JSON format  to
-           be applied to the started device.
+              definition,  a new transient device will be started.  If the UUID is omitted, a new UUID will be auto‐
+              matically generated. When starting a new transient device, the parent and device type must  be  speci‐
+              fied.   A  --jsonfile  may  replace the --type specification and also include additional attributes in
+              JSON format to be applied to the started device.
 
        stop DEVICESPEC
            Stop an mdev device, specified via its UUID.
 
-       types
-           List  the mdev device types known to the system by parent device.  Output may be limited to a single par‐
-           ent device with the -p|--parent option.  JSON output format is used with the --dumpjson option.
+       types  List the mdev device types known to the system by parent device.  Output may be limited  to  a  single
+              parent device with the -p|--parent option.  JSON output format is used with the --dumpjson option.
 
        undefine DEVICESPEC
-           Undefine, or remove the configuration for an mdev device, specified by its UUID and optionally  its  par‐
-           ent.  If  a  UUID  exists for multiple parents, all of them will be removed unless restricted to a single
-           parent.  Running devices are unaffected by this command.
+              Undefine,  or  remove  the  configuration for an mdev device, specified by its UUID and optionally its
+              parent. If a UUID exists for multiple parents, all of them will be removed unless restricted to a sin‐
+              gle parent.  Running devices are unaffected by this command.
 
 NOTE ON DEVICE SPECIFICATION
        For a given UUID, only one device with that UUID may be running at the same time. However, it is possible  to
@@ -274,7 +277,7 @@
            @{2}: {"assign_domain":"0xff"}
 
 CONFIGURATION FILE FORMAT
-       Configuration files are in JSON. Attributes in "attrs" are optional.
+       Configuration files are in JSON. Attributes in attrs are optional.
 
        {
          "mdev_type": "TYPE",
@@ -297,46 +300,45 @@
 
        Essentially, the procedure in mdevctl looks like this:
 
-              1. command-line parsing & setup
+          • command-line parsing & setup
 
-              2. invoke pre-command call-out
+          • invoke pre-command call-out
 
-              3. primary command execution*
+          • primary command execution [1]
 
-              4. invoke post-command call-out*
+          • invoke post-command call-out [1]
 
-              5. invoke notifier
+          • invoke notifier
 
-              * skipped if step 2 fails.
+          [1]  skipped if step 2 fails.
 
    EVENT SCRIPTS
-       A  call-out or notification event invokes a script along with a set of parameters detailing the type of call-
-       out, mdevctl's command execution progress, and the mediated device. The parameters are as follows:
+       A call-out or notification event invokes a script along with a  set  of  parameters  detailing  the  type  of
+       call-out, mdevctl's command execution progress, and the mediated device. The parameters are as follows:
 
        <CONFIG> | SCRIPT <-t=type -e=event -a=action -s=state -u=UUID -p=parent>
 
-       CONFIG
-           The device's JSON configuration, provided via standard input.
+       CONFIG The device's JSON configuration, provided via standard input.
 
        -t=type
            The device type.
 
        -e=event
-           Event type of call-out that is invoked. For call-out scripts, this may be pre, post, or get. For  notifi‐
-           cation scripts, this will always be notify.
+              Event type of call-out that is invoked. For call-out scripts, this may be pre, post, or get. For noti‐
+              fication scripts, this will always be notify.
 
        -a=action
            An action synonymous with an mdevctl command (e.g. define, start).
 
        -s=state
-           A  trinary  state of the mdevctl command execution. The possibilities are none if the mdevctl command has
-           yet to execute, success if the mdevctl command completed successfully, or failure if there was a  problem
-           executing the mdevctl command.
+              A trinary state of the mdevctl command execution. The possibilities are none if  the  mdevctl  command
+              has  yet  to execute, success if the mdevctl command completed successfully, or failure if there was a
+              problem executing the mdevctl command.
 
        -u=UUID
            UUID of the mediated device.
 
-       -p=parent>
+       -p=parent
            Parent of the mediated the device.
 
    CALL-OUT EVENT SCRIPTS
@@ -350,7 +352,6 @@
        get call-out events for the device type.
 
        Pre-Command
-
            A pre-command call-out event is invoked once prior to primary command execution.  Event type is pre. Sta‐
            tus will always be none.
 
@@ -362,10 +363,9 @@
            This event is not supported for the list, types, or version commands.
 
        Post-Command
-
-           A post-command call-out event is invoked once after primary command execution.  Event type is post.  Sta‐
-           tus will be success if mdevctl was able to finish primary command execution successfully, or failure oth‐
-           erwise.
+          A post-command call-out event is invoked once after primary command execution.  Event type is post. Status
+          will be success if mdevctl was able to finish primary command execution successfully,  or  failure  other‐
+          wise.
 
            The same script used for the pre event is used for the post event.
 
@@ -376,14 +376,13 @@
            This event is not supported for the list, types, or version commands.
 
        Get-attributes
-
            A get event is invoked during a define and list command to acquire device attributes from an  active  de‐
            vice.  Event type is get. Action is attributes. Status is none. Note that, unlike other call-outs events,
            get-attributes does not expect a device config on stdin, and an array of JSON formatted device attributes
            is returned via stdout.
 
-           The  same  script used for the pre event is used for the get event. If the script is not designed to sup‐
-           port a get event, then the return code is 0.
+          The same script used for the pre event is used for the get event. If the script is not designed to support
+          a get event, then the return code is 0.
 
            For define, a non-zero return code (exempting 2) will disrupt the define command entirely.
 
@@ -417,7 +416,6 @@
        command to other programs or loggers. Unlike call-out scripts, notifier scripts are device-type agnostic.
 
        Notify
-
            A notification event is invoked once either following a pre-command call-out failure or after a post-com‐
            mand  call-out.  Event is notify. If following a pre event, then status will be none. If following a post
            event, then status will mirror the value passed to the post-command call-out.
@@ -430,11 +428,13 @@
            This event is not supported for the list, types, or version commands.
 
    SCRIPT RETURN VALUES
-              0  if OK,
+       A call-out script should return one of the following values:
+
+          • 0  if OK,
 
-              1  if an error occurred,
+          • 1  if an error occurred,
 
-              2  if the script does not support the device type
+          • 2  if the script does not support the device type
 
 FILES
        /etc/mdevctl.d/*
@@ -450,6 +450,6 @@
        Scripts for notification call-out events.
 
 SEE ALSO
-       udev(7) udevadm(8) driverctl(8)
+       udev(7), udevadm(8), driverctl(8)
 
-                                                                                                          mdevctl(8)
+                                                                                                          MDEVCTL(8)
```

File: mdevctl.rst [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+    e2e73122-cc39-40ee-89eb-b0a47d334cae matrix vfio_ap-passthrough manual
+      Attrs:
+        @{0}: {"assign_adapter":"5"}
+        @{1}: {"assign_domain":"0x47"}
+        @{2}: {"assign_domain":"0xff"}
+
+CONFIGURATION FILE FORMAT
+=========================
+
+Configuration files are in JSON. Attributes in ``attrs`` are optional. ::
```
Comment (cohuck) [inline mdevctl.rst:327]: Minor: this seems to lose the quotes around "attrs" in the generated manpages. If there's no nice way to get them back, we can also leave it that way.

**Comment by cohuck** [comment]
This looks good to me!

File: mdevctl.rst (hunk 1 above)
Comment (jonner) [reply mdevctl.rst:327]: Oh, indeed. I don't think this was intentional. This should work:

```suggestion
Configuration files are in JSON. Attributes in "``attrs``" are optional. ::
```

File: build.rs [hunk 2]
```diff
@@ -31,6 +32,16 @@ fn apply_template(template: &Path) -> String {
         )
 }
 
+fn generate_manpage<P: AsRef<Path>>(outdir: P) {
+    let infile = PathBuf::from("mdevctl.rst");
+    let outfile = outdir.as_ref().join("mdevctl.8");
```
Comment (jonner) [inline build.rs:37]: It looks like I forgot the following line to make sure that the manpage gets re-generated if the input rst file is changed:

```rust
    println!("cargo:rerun-if-changed={}", infile.to_str().unwrap());
```

**Pull request merged**